### PR TITLE
hotfix: properly handle purl encoding

### PR DESCRIPTION
**Pull request by steven-esser**
* remove unneed exception raising for the version param
* do not skip URL encoding `:` and `+` characters
* add test cases from old issues

refs: #45, #46, #57

### CODE REVIEW

File: test/data/test-suite-data.json [hunk 1]
```diff
@@ -122,7 +122,7 @@
   {
     "description": "docker uses qualifiers and hash image id as versions",
     "purl": "pkg:docker/customer/dockerimage@sha256:244fd47e07d1004f0aed9c?repository_url=gcr.io",
-    "canonical_purl": "pkg:docker/customer/dockerimage@sha256:244fd47e07d1004f0aed9c?repository_url=gcr.io",
+    "canonical_purl": "pkg:docker/customer/dockerimage@sha256%3A244fd47e07d1004f0aed9c?repository_url=gcr.io",
```
Comment (matt-phylum) [inline test/data/test-suite-data.json]: The official test input says this should be unencoded (':').

I recommend keeping test-suite-data.json exactly the same as in the spec repository and making a separate file for tests that are unique to this repository. That makes it easier to ensure this implementation matches the spec, easier to update the copy of the test suite file, and easier to find tests that could be added to the spec.

Comment (steven-esser) [reply test/data/test-suite-data.json]: Good point.

**Comment by chrsch-dev** [comment]
Any updates when this will be merged?

**Comment by steven-esser** [comment]
@matt-phylum I am unsure what consensus was reach w.r.t. percent-encoding of `+` in the version field.

**Comment by stevehipwell** [comment]
Is there an ETA on getting this in?

**Comment by jdalton** [comment]
Closing as covered by https://github.com/package-url/packageurl-js/pull/71

**Comment by jdalton** [comment]
[v2.0.0](https://www.npmjs.com/package/packageurl-js/v/2.0.0) has been published 🎉

**Pull request closed without merging**